File: Subarray_Sum/min_subarray.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_len = len(nums) + 1
        sum = 0
        start = 0
        end = 0

        for num in nums:
            sum += num

            while sum >= target:
                if min_len > end - start + 1:
                    min_len = end - start + 1

                sum -= nums[start]
                start += 1

            end += 1

        if min_len == len(nums) + 1:
            return 0

        return min_len

File: Subarray_Sum/min_subarray.go
package subarraysum

func minSubArrayLen(target int, nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	minLen := len(nums) + 1
	sum := 0
	start := 0

	for end := 0; end < len(nums); end++ {
		sum += nums[end]

		for sum >= target {
			if minLen > end-start+1 {
				minLen = end - start + 1
			}

			sum -= nums[start]
			start++
		}
	}

	if minLen == len(nums)+1 {
		return 0
	}

	return minLen
}
